apex/contrib/csrc/groupbn/dnn.h

```diff
@@ -0,0 +1,26 @@
+#ifndef DNN_H
+#define DNN_H
+
+#ifdef __HIP_PLATFORM_HCC__
+#include <miopen/miopen.h>
+#define DNN_STATUS_SUCCESS miopenStatusSuccess
+#define DNN_DATA_HALF miopenHalf
+#define DNN_TENSOR_FORMAT 0
+
+using dnnTensorFormat_t = int;
+using dnnDataType_t = miopenDataType_t;
+using dnnStatus_t = miopenStatus_t;
+using dnnTensorDescriptor_t = miopenTensorDescriptor_t;
+#else
+#include <cudnn.h>
+#define DNN_STATUS_SUCCESS CUDNN_STATUS_SUCCESS
+#define DNN_DATA_HALF CUDNN_DATA_HALF
+#define DNN_TENSOR_FORMAT CUDNN_TENSOR_NHWC
+
+using dnnTensorFormat_t = cudnnTensorFormat_t;
+using dnnDataType_t = cudnnDataType_t;
+using dnnStatus_t = cudnnStatus_t;
+using dnnTensorDescriptor_t = cudnnTensorDescriptor_t;
+#endif
```

Comment:
Ideally we would want hipify to handle all of these instead of introducing a new file. @jithunnair-amd What do you think?

Reply:
I think most of the above can be added to the hipify mappings file, except a couple like `CUDNN_TENSOR_NHWC`->`0` and `cudnnTensorFormat_t`->`int`. @sarunyap Could you please elaborate on these two mappings? Are they done because MIOpen doesn't have any equivalent types? Would these mappings be correct in the general case or only for the use-case here?